packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -140,15 +140,20 @@ class _SnackBarActionState extends State<SnackBarAction> {
     final SnackBarThemeData snackBarTheme = Theme.of(context).snackBarTheme;
 
     MaterialStateColor resolveForegroundColor() {
-      if (widget.textColor is MaterialStateColor) {
-        return widget.textColor! as MaterialStateColor;
-      }
-      if (snackBarTheme.actionTextColor is MaterialStateColor) {
-        return snackBarTheme.actionTextColor! as MaterialStateColor;
-      }
-      if (defaults.actionTextColor is MaterialStateColor) {
-        return defaults.actionTextColor! as MaterialStateColor;
+      if (widget.textColor != null) {
```

Comment:
Great stuff; I think this is the logic we want. 

Can we add a unit test for this, in `snack_bar_test.dart`?

Reply:
I've added two tests. Sadly I'm not sure how to implement the case with `null` - Therefore I need the default color.
But I hope this sufficient and helpful enough.

Reply:
This is fine - I think the `null` case is covered sufficiently by other tests.